capstone/src/main/java/com/google/sps/search/servlets/NameDataServlet.java

```diff
@@ -0,0 +1,78 @@
+package com.google.sps.search.servlets;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.sps.search.SearchPredictor;
+import com.google.sps.servlets.AuthenticatedServlet;
+import com.google.sps.servlets.ServletHelper;
+
+@WebServlet("/name-data")
+public class NameDataServlet extends AuthenticatedServlet {
+
+  SearchPredictor searchPredictor;
+  String trieFile = "../../data/trie";
+
+  @Override
+  public void init(ServletConfig config) throws ServletException {
+    super.init(config);
+    try {
+      FileInputStream fileInput = new FileInputStream(new File(trieFile));
+      ObjectInputStream objectInput = new ObjectInputStream(fileInput);
+      searchPredictor = (SearchPredictor) objectInput.readObject();
+
+      fileInput.close();
+      objectInput.close();
+      return;
```

Comment:
If you return here, you don't get to line 45: 
`searchPredicctor = new SearchPredictor();`
I don't think that's intended?

Reply:
I initialize searchPredictor because it there's nothing to read it'll initialize the new object

Reply:
I've moved it above the try and got rid of the return